src/components/ListHeader/ListHeader.js

```diff
@@ -1,26 +1,94 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faShareAlt } from '@fortawesome/free-solid-svg-icons';
+import { faShareAlt, faCopy, faPencilAlt, faClock, faTrash } from '@fortawesome/free-solid-svg-icons';
 import { ReactComponent as Logo } from '../../static/logo.svg';
 
 import styles from './styles.module.css';
 
-const ListHeader = () => (
-  <div className={styles.listHeader}>
-    <div className={styles.headerRow}>
-      <Logo className={styles.logo} title="Logo" />
-      <FontAwesomeIcon
-        icon={faShareAlt}
-        className={styles.shareIcon}
-        title="Share this list"
-        tabIndex={0}
-      />
-    </div>
-    <div className={styles.headerRow}>
-      <h3>Shopping List</h3>
-      <h4>15 items</h4>
+const ListHeader = () => {
+  const [listDetailsShown, showListDetails] = useState();
+
+  const shareList = () => {
+    if (navigator.share) {
+      navigator.share({
+        title: 'Share Your List!',
+        url: 'https:thelist.app',
+      });
+    } else {
+      const listText = document.getElementById('listText');
+      listText.select();
+      document.execCommand('copy');
+    }
+  };
+
+  return (
+    <div className={styles.listHeader}>
+      <div className={styles.headerRow}>
+        <Logo className={styles.logo} title="Logo" />
+        <FontAwesomeIcon
+          icon={faShareAlt}
+          className={styles.shareIcon}
+          title="Share this list"
+          tabIndex={0}
+          onClick={() => showListDetails(!listDetailsShown)}
+        />
+      </div>
+      <div className={styles.headerRow}>
+        <h3>
+          Shopping List
+          <FontAwesomeIcon
+            icon={faPencilAlt}
+            className={styles.pencilIcon}
+            title="Edit list name"
+            tabIndex={0}
```

Comment:
In terms of the UX, what about this icon toggling `listDetailsShown` and the share icon just immediately calling `shareList`?

Reply:
That could work, would be a very easy swap too.

I felt like the share icon is definitely not a clear button to get the list details, this should be more intuitive!